### PR TITLE
For #12364 - Only show PWA prompt the third time a user visits instal…

### PR DESCRIPTION
**Pull request by ekager**
…lable site

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/perf/Performance.kt [hunk 1]
```diff
@@ -74,6 +74,6 @@ object Performance {
      * Disables the first time PWA popup.
      */
     private fun disableFirstTimePWAPopup(context: Context) {
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/perf/Performance.kt:76]:
```suggestion
    private fun disableThirdTimePWAPopup(context: Context) {
```
or
```suggestion
    private fun disablePwaOnboardingPopup(context: Context) {
```

**Comment by ekager** [comment]
Yeah, good call!

File: app/src/main/java/org/mozilla/fenix/shortcut/PwaOnboardingDialogFragment.kt [hunk 2]
```diff
@@ -18,7 +18,7 @@ import org.mozilla.fenix.ext.requireComponents
 /**
  * Dialog displayed the first time the user navigates to an installable web app.
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/shortcut/PwaOnboardingDialogFragment.kt:19]:
```suggestion
 * Dialog displayed the third time the user navigates to an installable web app.
```

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 3]
```diff
@@ -146,9 +147,18 @@ class Settings private constructor(
 
     // If any of the prefs have been modified, quit displaying the fenix moved tip
     fun shouldDisplayFenixMovingTip(): Boolean =
-        preferences.getBoolean(appContext.getString(R.string.pref_key_migrating_from_fenix_nightly_tip), true) &&
-            preferences.getBoolean(appContext.getString(R.string.pref_key_migrating_from_firefox_nightly_tip), true) &&
-            preferences.getBoolean(appContext.getString(R.string.pref_key_migrating_from_fenix_tip), true)
+        preferences.getBoolean(
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:150]: Looks like this is just a whitespace change, should it be removed?

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 4]
```diff
@@ -236,10 +246,10 @@ class Settings private constructor(
 
     val isCrashReportingEnabled: Boolean
         get() = isCrashReportEnabledInBuild &&
-            preferences.getBoolean(
-                appContext.getPreferenceKey(R.string.pref_key_crash_reporter),
-                true
-            )
+                preferences.getBoolean(
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:249]: same here

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt (hunk 3 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/utils/Settings.kt:150]: I agree it looks a bit ugly but this is what the autoformatter with the Kotlin style guide wants 🤷‍♀️ Don't have strong preferences but anytime you reformat this file you'd run into this

**Pull request merged**